### DISCUSSION TITLE
Batch Job didn't wait delayed time

### DISCUSSION THREAD

**Discussion by mohamedali-pelcro** [laravel/framework#38990] [Q&A]
- Laravel Version: ^8
- PHP Version: 7.4
- Database Driver & Version: Mysql 8 

### Description:
$jobs[] =  (new FirstJob)->delay(now()->addSeconds(90));
$jobs[] =  (new SecondJob)->delay(now()->addSeconds(120));

Bus::batch([$jobs])->->dispatch();

the first job dispatched without waiting to it's delayed time

**Comment by tsybenko** [comment]
Laravel uses "sync" queue driver by default. Did you try to run it with some another driver?

**Comment by mohamedali-pelcro** [comment]
> Laravel uses "sync" queue driver by default. Did you try to run it with some another driver?

Yes I tried with redis

the issue here they add them as a chain but the first one dispatched first so is there any way to add to chain instead of dispatch it ?

[https://github.com/laravel/framework/blob/6216534f14c7cb8bdfa213e5f7e3d357119a4579/src/Illuminate/Bus/Batch.php#L161-L197](https://github.com/laravel/framework/blob/6216534f14c7cb8bdfa213e5f7e3d357119a4579/src/Illuminate/Bus/Batch.php#L161-L197)

**Reply by mohamedali-pelcro** [reply]
Does anyone have a solution for this?

**Reply by woodspire** [reply]
Shouldn't

Bus::batch([$jobs])->->dispatch();

be

Bus::batch($jobs)->->dispatch();

instead ?

$jobs is already an array. No need to wrap it in another array, no ?
